Remove legacy VPN solution (#7167)
gardener/gardener: docs/usage/reversed-vpn-tunnel.md

# Reversed VPN Tunnel Setup and Configuration 

The Reversed VPN Tunnel is enabled by default.

> A high availability VPN connection is automatically deployed in all shoots that configure an HA control-plane.

Reviewer: ```suggestion
A highly available VPN connection is automatically deployed in all shoots that configure an HA control-plane.
```

Author: Fixed.

> ## Reversed VPN Tunnel

Reviewer: ```suggestion
A high availability VPN connection is automatically deployed in all shoots that configure an HA control-plane.

## Reversed VPN Tunnel
```

Author: Fixed.

In the first VPN solution, connection establishment was initiated by a VPN client in the seed cluster.
Due to several issues with this solution, the tunnel establishment direction has been reverted.
The client is deployed in the shoot and initiates the connection from there. This way, there is no need to deploy a special purpose
loadbalancer for the sake of addressing the data-plane, in addition to saving costs, this is considered the more secure alternative.
For more information on how this is achieved, please have a look at the following [GEP](../proposals/14-reversed-cluster-vpn.md).

Connection establishment with a reversed tunnel:

`APIServer --> Envoy-Proxy | VPN-Seed-Server <-- Istio/Envoy-Proxy <-- SNI API Server Endpoint <-- LB (one for all clusters of a seed) <--- internet <--- VPN-Shoot-Client --> Pods | Nodes | Services`

The reversed VPN tunnel is always deployed.

> The feature gate `ReversedVPN` is GA and will be removed in a future release

Reviewer: ```suggestion
The feature gate `ReversedVPN` is GA and will be removed in a future release.
```
